src/io/record-play-movie.cpp

```diff
@@ -277,17 +321,16 @@ void prepare_chuukei_hooks(void)
  */
 void prepare_movie_hooks(PlayerType *player_ptr)
 {
-    char buf[1024];
-    char tmp[80];
-
     if (movie_mode) {
         movie_mode = 0;
         disable_chuukei_server();
         fd_close(movie_fd);
         msg_print(_("録画を終了しました。", "Stopped recording."));
     } else {
-        sprintf(tmp, "%s.amv", player_ptr->base_name);
+        char tmp[80];
+        strnfmt(tmp, sizeof(tmp), "%s.amv", player_ptr->base_name);
```

Comment:
This code can be changed below:
```cpp
const auto filename = format("%s.amv", player_ptr->base_name);
if (get_string(_("ムービー記録ファイル: ", "Movie file name: "), filename.data(), 80)) {
```

Reply:
The latest forced push adopts your suggestion, but I dropped `const` since get_string() can modify the contents and the gcc 10.2.1 on Linux flagged the get_string() line with an error when `const` was included.

Reply:
When using std::string::data(), which is not a const member function, the following caveat applies.

> Modifying the past-the-end null terminator stored at data()+size() to any value other than CharT() has undefined behavior. (https://en.cppreference.com/w/cpp/string/basic_string/data)

Thus, specifying a filename longer than the default filename in the get_string function will result in undefined behavior.
I think we have to use the method of strnfmt to write the default filename to the char array buffer as before.

Reply:
The latest forced push reverts to strnfmt() to fill in the variable passed to get_string().  Did change the name of that variable from 'tmp' to 'filename'.